components/infobox/commons/infobox_widget_injector.lua

```diff
@@ -9,7 +9,14 @@
 local Class = require('Module:Class')
 
 ---@class WidgetInjector
-local Injector = Class.new()
+---@operator call(table?): WidgetInjector
+---@field caller table?
+local Injector = Class.new(
+	---@param self self
+	---@param caller table?
+	function(self, caller)
```

Comment:
Since this is an `infobox_widget_injector` we might as well do 
```suggestion
	function(self, infobox)
```
Sorry for not noticing this in the first suggestion ^^

Reply:
Alternative could be to use `args` and pass just these

Reply:
https://github.com/Liquipedia/Lua-Modules/blob/33d11eb470fd83f824af1fc601aefddf48bf49c9/components/prize_pool/commons/prize_pool_base.lua#L22

it is used in prize pool too

as for just passing args: that would make it not possible to kick stuff like `_league`

Reply:
Is _league really used for other purposes than accessing args?

Reply:
yes, e.g. to access functions from the basic infobox

Reply:
getAllArgsForBase should be made a static method instead of instance method, which would solve that issue also

Reply:
should we move it to module:Table?
could be usable in non infoboxes then too

Reply:
as for additional examples where we would want self (the caller) instead of args available:
- have self.pagename available in the injector (several infoboxes on several wikis)
- have `self.totalEarnings` and `self.earningsPerYear` available in the injector in person/team infoboxes (earnings current year display as an example)

Reply:
why i do not like calling it infobox:
- The injector being used in prize pool module too (if we would want to access the caller there it would imo be awkward to use `self.infobox` in prize pool modules)
- Technically it isn't the infobox available there but the custom/base of the infobox specification (for lack of a better description); there would be `self.infobox.infobox` then

if there are other suggestions for the name i am all ears though^^